runtime/master/src/main/java/edu/snu/onyx/runtime/master/RuntimeMaster.java

```diff
@@ -165,20 +169,29 @@ public void terminate() {
   }
 
   public void requestContainer(final String resourceSpecificationString) {
-    try {
-      final TreeNode jsonRootNode = objectMapper.readTree(resourceSpecificationString);
-
-      for (int i = 0; i < jsonRootNode.size(); i++) {
-        final TreeNode resourceNode = jsonRootNode.get(i);
-        final ResourceSpecification.Builder builder = ResourceSpecification.newBuilder();
-        builder.setContainerType(resourceNode.get("type").traverse().nextTextValue());
-        builder.setMemory(resourceNode.get("memory_mb").traverse().getIntValue());
-        builder.setCapacity(resourceNode.get("capacity").traverse().getIntValue());
-        final int executorNum = resourceNode.path("num").traverse().nextIntValue(1);
-        containerManager.requestContainer(executorNum, builder.build());
+    final Future<?> containerRequestEventResult = masterControlEventExecutor.submit(() -> {
+      try {
+        final TreeNode jsonRootNode = objectMapper.readTree(resourceSpecificationString);
+
+        for (int i = 0; i < jsonRootNode.size(); i++) {
+          final TreeNode resourceNode = jsonRootNode.get(i);
+          final ResourceSpecification.Builder builder = ResourceSpecification.newBuilder();
+          builder.setContainerType(resourceNode.get("type").traverse().nextTextValue());
+          builder.setMemory(resourceNode.get("memory_mb").traverse().getIntValue());
+          builder.setCapacity(resourceNode.get("capacity").traverse().getIntValue());
+          final int executorNum = resourceNode.path("num").traverse().nextIntValue(1);
+          resourceRequestCount.getAndAdd(executorNum);
+          containerManager.requestContainer(executorNum, builder.build());
+        }
+      } catch (final Exception e) {
+        throw new ContainerException(e);
       }
-    } catch (final IOException e) {
-      throw new RuntimeException(e);
+    });
+    try {
+      containerRequestEventResult.get();
+    } catch (final Exception e) {
+      e.printStackTrace();
```

Comment:
Why don't we use `Logger` like [this](https://stackoverflow.com/questions/4347797/how-to-send-a-stacktrace-to-log4j)?